cpp/src/ray/runtime/abstract_ray_runtime.h

```diff
@@ -0,0 +1,71 @@
+
+#pragma once
+
+#include <mutex>
+
+#include <ray/api/ray_config.h>
+#include <ray/api/ray_runtime.h>
+#include <ray/core.h>
+#include <msgpack.hpp>
+#include "./object/object_store.h"
+#include "./task/task_executer.h"
+#include "./task/task_submitter.h"
+
+namespace ray {
+namespace api {
+
+class AbstractRayRuntime : public RayRuntime {
+  friend class Ray;
+
+ private:
+ protected:
+  static std::unique_ptr<AbstractRayRuntime> ins_;
```

Comment:
Why we need this static field? We already have the static `impl_` field in `Ray` class.

Reply:
Provide `AbstractRayRuntime` used by runtime modules internally. `AbstractRayRuntime` has many internal method relative to RayRuntime.